### PR TITLE
Delete NS2x runtime specific impls

### LINKED ISSUE #64536: Supporting netstandard2.x platform specific target frameworks is causing unreasonable pain
The toolchain (mostly NuGet) supports platform specific target frameworks since net5.0 but .NETStandard specific platform target frameworks never were and likely never will be supported (we asked the NuGet team and they refused because of technical limitations). During the migration from the old configuration system (which was `TargetGroup` based) to the Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk package (`TargetFrameworks` based) we had to trick NuGet to not see target framework platform overloads so that the toolchain doesn't raise errors. At that time, netstandard2.x wasn't the only tfm with a platform overload, others were `netcoreapp2.x-`, `netcoreapp3.x-`, `netstandard1.x-`, `net46x-`. We had to apply the mentioned trick (or better to say, hack) on every target framework in the `TargetFrameworks` property.

Nowadays, the only by the toolchain unsupported platform specific target frameworks still in use in our libraries are `netstandard2.0` and `netstandard2.1`. The following libraries still make use of them:

- Microsoft.Win32.Registry.AccessControl: `netstandard2.0-windows;netstandard2.0`
- System.Data.OleDb: `netstandard2.0-windows;netstandard2.0`
- System.IO.Ports: `netstandard2.0-windows;netstandard2.0-Unix;netstandard2.0`
- System.Net.Http.WinHttpHandler: `netstandard2.1-windows;netstandard2.1;netstandard2.0-windows;netstandard2.0`
- System.Runtime.Caching: `netstandard2.0-windows;netstandard2.0`
- System.Security.Cryptography.Pkcs: `netstandard2.1-windows;netstandard2.1;netstandard2.0-windows;netstandard2.0`
- System.Security.Cryptography.ProtectedData: `netstandard2.0-windows;netstandard2.0`
- System.ServiceProcess.ServiceController: `netstandard2.0-windows;netstandard2.0`
- System.Text.Encoding.CodePages: `netstandard2.0-windows;netstandard2.0`
- System.Threading.AccessControl: `netstandard2.0-windows;netstandard2.0`

As .NETStandard will continue to be supported by libraries in dotnet/runtime for many years (just my own guess - nothing official), I would like to propose removing the `netstandard2.x` platform specific target frameworks, which aren't supported by the toolchain, from the above list.

The reason why I feel strongly about removing `netstandard2.x` platform specific target frameworks is because of the following inconveniences that exist because of the hack to satisfy the toolchain:
- Visual Studio or any other toolchain that relies on the [project-system](https://github.com/dotnet/project-system) can't restore solutions with platform specific netstandard2.x projects. Instead the restore must be intentionally disabled via a NuGet.config file per solution file, i.e. https://github.com/dotnet/runtime/blob/215c328e039b3a6b6bfdc6514253307105f25816/src/libraries/System.DirectoryServices.AccountManagement/NuGet.config#L2
That hack is only supported by static graph restore. A normal/legacy restore without static graph will blow up. Besides, any external component which isn't considered by the hack won't be able to interpret the overloaded target frameworks either. In general, the number of issues that have been opened because of IDEs not being able to interpret the overloaded target frameworks, strongly indicate unreasonable pain.
- The sdk's `AppendTargetFrameworkToOutputPath` feature which is on by default had to be disabled and instead, `OutputPath` and `IntermediateOutputPath` had to be set manually to honor the stripped "suffix".
- TargetPlatformIdentifier conditions, which is the recommended way to condition based on the platform specific tfms, can't be used in combination with netstandard2.x. In addition to that, the SDK defaults netstandard2.x's platform to Windows. As the TargetFramework "suffix" is stripped from netstandard2.x, we need to rely on a custom property `TargetFrameworkSuffix` to form conditions that should only apply to either the platform specific or the platform agnostic netstandard2.x tfm whenever both are present. This makes the already complex item conditions in our libraries even harder to read and write.

Alternatives like runtime checks which don't rely on platform specific `netstandard2.x` target frameworks should be explored, especially because of the number of projects that still use them is very low.

@ericstj @safern

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/64536

Remove the remaining ten .NETStandard runtime specific implementations.
For reasoning please see #64536.

@ericstj I'm unsure if I got it right. Some of the NS2.x implementations will now throw a PNSE. Is that expected and OK? **All the project changes are contained in the first commit**. The second commit removes the infrastructure burden.

### CODE REVIEW

**Comment by ericstj** [comment]
> Some of the NS2.x implementations will now throw a PNSE. Is that expected and OK? 

So long as that's what they did before this change.  We don't want to transition any `netstandard` implementation from real to PNSE without first considering all scenarios.  I don't see any of those transitions here so it seems OK.

**Comment by ViktorHofer** [comment]
Thanks for double checking. Then the change is ready to be reviewed. I believe the test leg failures are all unrelated.

File: src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj [hunk 1]
```diff
@@ -20,7 +20,7 @@
     
     <_generateRuntimeGraphTargetFramework Condition="'$(MSBuildRuntimeType)' == 'core'">$(NetCoreAppToolCurrent)</_generateRuntimeGraphTargetFramework>
     <_generateRuntimeGraphTargetFramework Condition="'$(MSBuildRuntimeType)' != 'core'">net472</_generateRuntimeGraphTargetFramework>
-    <_generateRuntimeGraphTask>$([MSBuild]::NormalizePath('$(BaseOutputPath)', '$(_generateRuntimeGraphTargetFramework)-$(Configuration)', '$(AssemblyName).dll'))</_generateRuntimeGraphTask>
+    <_generateRuntimeGraphTask>$([MSBuild]::NormalizePath('$(BaseOutputPath)', $(Configuration), '$(_generateRuntimeGraphTargetFramework)', '$(AssemblyName).dll'))</_generateRuntimeGraphTask>
```
Comment (ericstj) [inline src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj:23]: How is this related to the rest of the change?

File: eng/testing/tests.props [hunk 2]
```diff
@@ -24,10 +24,9 @@
 
   <PropertyGroup Condition="'$(TargetsMobile)' == 'true'">
     <!-- Test runners are built as part of libs.pretest so we need to use libraries configuration -->
-    <MobileRunnersDirSuffix>$(NetCoreAppCurrent)-$(Configuration)</MobileRunnersDirSuffix>
-    <AppleTestRunnerDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'AppleTestRunner', '$(MobileRunnersDirSuffix)'))</AppleTestRunnerDir>
-    <AndroidTestRunnerDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'AndroidTestRunner', '$(MobileRunnersDirSuffix)'))</AndroidTestRunnerDir>
-    <WasmTestRunnerDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'WasmTestRunner', '$(MobileRunnersDirSuffix)'))</WasmTestRunnerDir>
+    <AppleTestRunnerDir>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'AppleTestRunner', '$(Configuration)', '$(NetCoreAppCurrent)'))</AppleTestRunnerDir>
```
Comment (ericstj) [inline eng/testing/tests.props:27]: How is this related to the rest of the change?

File: docs/coding-guidelines/libraries-packaging.md [hunk 3]
```diff
@@ -92,6 +92,3 @@ In order to mitigate design-time/build-time performance issues with source gener
     <DisableSourceGeneratorPropertyName>CustomPropertyName</DisableSourceGeneratorPropertyName> 
   </PropertyGroup>
 ```
```
Comment (ericstj) [inline docs/coding-guidelines/libraries-packaging.md:94]: Should we add something to the docs around where the `-runtime` suffix is supported?  Does the infra fail in a sensible way when it's misused (if not, should we explicitly error)?

**Review by ericstj — APPROVED** [review]
I went through everything and it looked reasonable.

File: docs/coding-guidelines/libraries-packaging.md (hunk 3 above)
Comment (ViktorHofer) [reply docs/coding-guidelines/libraries-packaging.md:94]: After https://github.com/dotnet/arcade/pull/8418 merged and consumed, NuGet will throw during restore if a non .NETCoreApp tfm contains a suffix. It makes sense to add section to the [project guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/project-guidelines.md) clarifying which tfms can be runtime specific and how to apply conditions to specific runtimes correctly.

I plan to update that document with https://github.com/dotnet/runtime/pull/64500.

File: src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj (hunk 1 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj:23]: I change the IntermediateOutputPath and the OutputPath by removing the `AppendTargetFrameworkToOutputPath=false` setting in the libraries Directory.Build.props file, hence a few hardcoded paths need to be updated.

**Comment by ViktorHofer** [comment]
The wasm failures are affecting all PRs. Merging.

**Review by safern — COMMENTED** [review]
LGTM
